packages/typescript-client/src/client.ts

```diff
@@ -388,6 +401,37 @@ export class ShapeStream {
       )
     }
   }
+
+  // Parses the message values using the provided parser based on the schema information
+  private parse(key: string, value: string): Value {
+    const columnInfo = this.schema?.[key]
+    // use guards for the "type" property
+    if (!columnInfo) {
+      // We don't have information about the value
+      // so we just return it
+      return value
+    }
+
+    // Pick the right parser for the type
+    const parser =
```

Comment:
nitpick: correct type here is

```suggestion
    const parser: ParseFunction | undefined =
```